### ISSUE TITLE
Implement reinforcement learning

### ISSUE THREAD

**Issue by Diapolo10** [Diapolo10/rl_car#8]
The heart of the project; actually implementing the neural network that drives the car.

This will first require deciding which library will be used for the neural network. The current candidates are PyTorch and Tensorflow, the former likely being a better fit, but the end result could go either way.

Next, we need to implement the neural network's feedback functions. Namely, what actions to reward and what to punish. The NN could be rewarded for completing consecutive laps, giving more for each loop the faster it completes it, though due to the complexity of the track we may need to add "checkpoints" along the map that give a small reward each. On the opposite end we obviously punish the neural network hard for crashing into the walls, and we may have to punish it for idling.

The input nodes should likely be:

- Vision lasers' intersection points (~12 in total)
- Vehicle x-axis velocity
- Vehicle y-axis velocity
- Vehicle angle

The output nodes should be:

- Left arrow (turn left)
- Right arrow (turn right)
- Up arrow (forward)
- Down arrow (brake/backward)

All of this is subject to change as we make further progress. All feedback is welcome.
